ports/esp32/boards/deploy_c2.md

```diff
@@ -0,0 +1,14 @@
+Program your board using the esptool.py program, found [here](https://github.com/espressif/esptool).
```

Comment:
A reminder for the future that #16535 deletes these copy-paste deploy.md files, so depending on which PR merges first there is a cleanup step for the PR which is merged second.

Reply:
The linked PR just merged, so it's OK to rebase this, remove this file, and link to the generic `deploy.md` now.